src/lib/jsonParseV2/parseRegistration.cpp

```diff
@@ -0,0 +1,367 @@
+  {
+    if (!parseStringVector(&dataProvidedP->attributes, dataProvided["attrs"], "attrs", true, errorStringP))
+    {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+
+
+/* ****************************************************************************
+*
+* httpParse -
+*/
+static bool httpParse(ngsiv2::Http* httpP, const rapidjson::Value& jsonHttp, const std::string& fieldName, std::string* errorStringP)
+{
+  if (!jsonHttp.IsObject())
+  {
+    *errorStringP = "/" + fieldName + "/ must be a JSON object";
+    return false;
+  }
+
+  if (!jsonHttp.HasMember("url"))
+  {
+    *errorStringP = "/" + fieldName + "/ must contain a field /url/";
+    return false;
+  }
+
+  const rapidjson::Value& url = jsonHttp["url"];
+
+  if (!url.IsString())
+  {
+    *errorStringP = "/url/ field of /" + fieldName + "/ must be a JSON string";
+    return false;
+  }
+
+  httpP->url = url.GetString();
+
+  if (httpP->url == "")
+  {
+    *errorStringP = "/url/ field of /" + fieldName + "/ cannot be the empty string";
+    return false;
+  }
+
+  std::string  host;
+  int          port;
+  std::string  path;
+  std::string  protocol;
+
+  if (parseUrl(httpP->url, host, port, path, protocol) == false)
+  {
+    *errorStringP = "/url/ field of /" + fieldName + "/ is invalid";
+    return false;
+  }
+
+  return true;
+}
+
+
+
+/* ****************************************************************************
+*
+* providerParse -
+*/
+static bool providerParse(ConnectionInfo* ciP, ngsiv2::Provider* providerP, const rapidjson::Value& provider, std::string* errorStringP)
+{
+  bool b;
+
+  if (!provider.IsObject())
+  {
+    *errorStringP = "/provider/ must be a JSON object";
+    return false;
+  }
+
+  if (provider.ObjectEmpty())
+  {
+    *errorStringP = "/provider/ empty";
+    return false;
+  }
+
+  if (provider.HasMember("http"))
+  {
+    b = httpParse(&providerP->http, provider["http"], "http", errorStringP);
+    if (b == false)
+    {
+      return false;
+    }
+  }
+  else
+  {
+    *errorStringP = "/provider/ must contain a field /http/";
+    return false;
+  }
+
+  if (provider.HasMember("supportedForwardingMode"))
+  {
+    const rapidjson::Value& supportedForwardingMode = provider["supportedForwardingMode"];
+    if (!supportedForwardingMode.IsString())
+    {
+      *errorStringP = "/supportedForwardingMode/ must be a JSON string";
+      return false;
+    }
+
+    providerP->supportedForwardingMode = supportedForwardingMode.GetString();
+    
+    //
+    // Valid strings: ngsiV2, legacy, all
+    //
+    if ((providerP->supportedForwardingMode != "ngsiV2") &&  // FIXME PR: what are the exact values?
+        (providerP->supportedForwardingMode != "legacy") &&
+        (providerP->supportedForwardingMode != "all"))
+    {
+      *errorStringP = "invalid value of /supportedForwardingMode/";
+      return false;
+    }
+  }
+  else
+  {
+    providerP->supportedForwardingMode = "legacy";  // FIXME PR: what is the default value?
```

Comment:
At the present moment the only possiblity (so "default") is legacy. However, maybe a comment about this would be changed in the future, once NGSIv2 forwarding gets defined would be good. However, not at FIXME PR level (it should point to the proper issue... not sure if we have some proper issue at the present moment. To be searched).

Reply:
ok
I made a "PR FIXME" as I had no idea what to do with this ...
Do we have a PR about APIv2 forwarding yet?

Reply:
[I understand you refer to issue, not PR]

Not really... What we have is just a side comment in a PR on other topic. Not very clear :). I'll grab that piece of information and create and issue about it. We are going to needed, as NGSIv2 implementation for IOTAs is also pending on it.

Reply:
Issue created: https://github.com/telefonicaid/fiware-orion/issues/3068

Reply:
Fixed in 5686ca28a